flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPool.java

```diff
@@ -212,7 +212,7 @@ public void start(JobMasterId jobMasterId, String newJobManagerAddress) throws E
 
 		// release all registered slots by releasing the corresponding TaskExecutors
 		for (ResourceID taskManagerResourceId : registeredTaskManagers) {
-			releaseTaskManagerInternal(taskManagerResourceId);
+			releaseTaskManagerInternal(taskManagerResourceId, null);
```

Comment:
I would try to avoid null here. It is still helpful to have a message that says "SlotPool is shutting down".